File: examples/python/11.poo/01.class/01-class_attr_methods.py
#!/usr/bin/env python3
import sys

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def greet(self):
        print("Hello! I am " + self.name)

    # Ruby use to_s method name
    def __str__(self):
        return "Name: " + self.name + ", Age: " + str(self.age)

p = Person('Yoda', 900) # Ruby call new method to build instance
p.greet()
print(p)                # Python call __str__ method to convert object to String

Person('Obiwan', 53).greet()

File: examples/2.high-level/python/11.poo/01.class/01-class_attr_methods.rb
#!/usr/bin/env ruby
class Person
  def initialize(name, age)
    @name = name
    @age = age
  end

  def greet
    puts("Hello! I'm #{@name}.")
  end

  def to_s
    "Name: #{@name}, Age: #{@age}"
  end
end

p = Person.new('Yoda', 900)
p.greet
puts p                      # Call method to_s to convert object to String

Person.new('Obiwan', 53).greet
